Packages/com.chisel.editor/Chisel/Editor/Editor/MaterialBrowser/ChiselMaterialBrowserUtilities.cs

```diff
@@ -0,0 +1,242 @@
+/* * * * * * * * * * * * * * * * * * * * * *
+Chisel.Editors.ChiselMaterialBrowserCache.cs
+
+License: MIT (https://tldrlegal.com/license/mit-license)
+Author: Daniel Cornelius
+
+* * * * * * * * * * * * * * * * * * * * * */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Chisel.Components;
+using UnityEditor;
+using UnityEngine;
+
+namespace Chisel.Editors
+{
+    internal static class ChiselMaterialBrowserUtilities
+    {
+        // checks a path and returns true/false if a material is ignored or not
+        public static bool IsValidEntry( ChiselMaterialBrowserTile tile )
+        {
+            // these are here to clean things up a little bit and make it easier to read
+
+            bool PathContains( string path )
+            {
+                return tile.path.ToLower().Contains( path );
+            }
+
+            // checks for any shaders we want to exclude
+            bool HasInvalidShader()
+            {
+                string shader = tile.shaderName.ToLower();
+
+                string[] excludedShaders = new string[]
+                {
+                        "skybox/"
+                };
```

Comment:
new'ing an array creates garbage. you could make this readonly static outside the method (since it never changes) or just use the first element since it only has one element at the moment

Reply:
Fixed this. No longer uses manually typed comparisons, and this method overall has been rewritten. Ideally I'll be adding a preference option so the end user can add their own ignored folders and assets. 
Will resolve once I push next commit.